### PR TITLE
fix test

### PR DESCRIPTION
**Pull request by vchiley**
found a few import errors and fixed a bunch of lint issues

### CODE REVIEW

**Review by mvpatel2000 — APPROVED** [review]
remove the type ignore?  is it still necessary

**Review by dakinggg — APPROVED** [review]

File: llmfoundry/common/convert_dataset.py [hunk 1]
```diff
@@ -300,6 +302,11 @@ def build_hf_dataset(dataset_name: str,
                                   data_subset=data_subset,
                                   split=split)
     else:
+        if not isinstance(tokenizer, PreTrainedTokenizerBase):
+            raise ValueError(
+                f'{tokenizer=} must be of type PreTrainedTokenizerBase')
+        if max_length == 0:
+            raise ValueError(f'max_length must be set.')
```
Comment (mvpatel2000) [inline llmfoundry/common/convert_dataset.py:309]: In this case, can we just not have it be a default arg of 0 and make it a required arg

**Review by mvpatel2000 — APPROVED** [review]
Many good catches. Thanks!

File: llmfoundry/common/fdiff.py [hunk 2]
```diff
@@ -25,39 +25,38 @@ def __init__(self, diff_train_metrics=False, diff_eval_metrics=True):
 
     def batch_end(self, state: State, logger: Logger):
         if self.diff_train_metrics:
+            loss = state.loss.item()  # type: ignore
```
Comment (hanlint) [inline llmfoundry/common/fdiff.py:28]: I'm guessing the type error here is because `state.loss` can either be a tensor or a list. In this case, the better approach is to throw an error here that multiple losses are not supported by this callback. e.g. see: https://github.com/mosaicml/composer/blob/dev/composer/algorithms/mixup/mixup.py#L203

File: llmfoundry/common/fdiff.py [hunk 3]
```diff
@@ -25,39 +25,38 @@ def __init__(self, diff_train_metrics=False, diff_eval_metrics=True):
...
-            self.train_prev_loss = state.loss.item()
+                logger.log_metrics(
+                    {'loss/train/total_fdiff': loss - self.train_prev_loss})
+            self.train_prev_loss = loss
 
             for k in self.train_prev_metric.keys():
                 logger.log_metrics({
                     f'metrics/train/{k}_fdiff':
-                        state.train_metric_values[k].item() -
+                        state.train_metric_values[k].item() -  # type: ignore
```
Comment (hanlint) [inline llmfoundry/common/fdiff.py:37]: The issue here is that `state.train_metric_values` is typed as a `Dict[str, float]` (https://github.com/mosaicml/composer/blob/dev/composer/core/state.py#L504), but this code assumes a tensor was put in. So is the `.item()` needed here? Is this masking an API misuse upstream?

File: llmfoundry/common/convert_dataset.py (hunk 1 above)
Comment (vchiley) [reply llmfoundry/common/convert_dataset.py:309]: if we take the `if mode == ConcatMode.NO_CONCAT:` path, then max_length is not required.

Comment (hanlint) [reply llmfoundry/common/convert_dataset.py:309]: small nit: in this case, its common to do `max_length: Optional[int] = None` , and then check for `None`. Avoids using a magic number (0), and makes the usage clear in the type annotation. Also noticed that `max_length` is missing from the docstrings.

File: llmfoundry/common/fdiff.py (hunk 3 above)
Comment (vchiley) [reply llmfoundry/common/fdiff.py]: This callback isn't used super often
I think you just found a bug
:thankyou:
